File: FindDisappearedNums.py
"""
Time Complexity : O(n)
Space Complexity : O(1)
Did this code successfully run on Leetcode : Yes
Any problem you faced while coding this : No
"""
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        if len(nums) == 0:
            return []
        result = []
        # Changing index numbers to '-' once found
        for i in range(len(nums)):
            index = abs(nums[i]) - 1
            if nums[index] > 0:
                nums[index] = nums[index] * -1
        # Finding the numbers with discrepancy i.e the 
        # positive numbers and incrementing their index and return them
        for i in range(len(nums)):
            if nums[i] > 0:
                result.append(i + 1)
        return result

File: FindDisappearedNums.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {

        List<Integer> list= new ArrayList<Integer>();

        for(int i=0; i< nums.length; i++)
        {
            int indx = 0;
            indx = Math.abs(nums[i]) - 1;   // number can be negative so take absolute
            if(nums[indx]>0) // If already made negative dont do anything probably the repeated number 
             nums[indx]*= -1; // if not negative make it negative
        }
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]>0 )     
            list.add(i+1); // any positive values? that means the element at that index is missing
            else
            nums[i]*= -1;  // restore original array
            
        }
        return list;
    }
}